awx/api/serializers.py

```diff
@@ -4157,7 +4157,10 @@ def validate(self, attrs):
         # verify that credentials (either provided or existing) don't
         # require launch-time passwords that have not been provided
         if 'credentials' in accepted:
-            launch_credentials = accepted['credentials']
+            launch_credentials = (
+                list(accepted['credentials']) +
```

Comment:
@AlanCoding this accepted list that comes back *only* includes _net new_ credentials - not ones that **already exist** on the JT prior to launch, so we end up looking at the wrong list of materialized credentials to determine _which passwords we need_.

Reply:
This might give duplicates inside of `launch_credentials`, but from reading the code after this carefully, I can't see any harm which would come from that.

Reply:
I changed this slightly to address this.